File: src/Neuralm.Services/Neuralm.Services.TrainingRoomService/Neuralm.Services.TrainingRoomService.Messages/GetTrainingRoomResponse.cs
using Neuralm.Services.Common.Messages.Abstractions;
using Neuralm.Services.TrainingRoomService.Messages.Dtos;

namespace Neuralm.Services.TrainingRoomService.Messages
{
    /// <summary>
    /// Represents the <see cref="GetTrainingRoomResponse"/> class.
    /// </summary>
    public class GetTrainingRoomResponse : Response
    {
        /// <summary>
        /// Gets and sets the training room.
        /// </summary>
        public TrainingRoomDto TrainingRoom { get; set; }
        
        /// <summary>
        /// Initializes an instance of the <see cref="GetTrainingRoomResponse"/> class.
        /// SERIALIZATION CONSTRUCTOR.
        /// </summary>
        public GetTrainingRoomResponse()
        {
            
        }
    }
}

File: src/Neuralm.Presentation.Web/src/messages/responses/GetTrainingRoomResponse.ts
import Response from './Response';
import TrainingRoom from '@/models/TrainingRoom';

/**
 * Represents the get enabled training rooms response class.
 */
export default class GetTrainingRoomResponse extends Response {
  public trainingRoom: TrainingRoom;

  /**
   * Initializes a new instance of the GetEnabledTrainingRoomsResponse class.
   */
  constructor(id: string, requestId: string, dateTime: Date, message: string, success: boolean, trainingRoom: TrainingRoom) {
    super(id, requestId, dateTime, message, success);
    this.trainingRoom = trainingRoom;
  }
}
